src/oscar/apps/dashboard/ranges/forms.py

```diff
@@ -30,9 +31,15 @@ class RangeProductForm(forms.Form):
     file_upload = forms.FileField(
         label=_("File of SKUs or UPCs"), required=False, max_length=255,
         help_text=_('Either comma-separated, or one identifier per line'))
+    upload_type = forms.CharField(widget=forms.HiddenInput(), required=False)
 
     def __init__(self, range, *args, **kwargs):
         self.range = range
+        # make upload_type accessible in clean_query
+        if kwargs.get("initial"):
```

Comment:
I cannot access cleaned_data['upload_type'] in clean_query() - the solution to avoid this would be to merge clean_query() into clean()